master/internal/run/runs.go

```diff
@@ -0,0 +1,178 @@
+package run
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const (
+	integerType   = "integer"
+	floatType     = "float"
+	booleanType   = "boolean"
+	stringType    = "string"
+	timestampType = "timestamp"
+	// MaxMetadataValueStringLength is the maximum length of a metadata value.
+	MaxMetadataValueStringLength = 5000
+	// MaxMetadataKeyLength is the maximum length of a metadata key.
+	MaxMetadataKeyLength = 50
+	// MaxMetadataArrayLength is the maximum length of a metadata array.
+	MaxMetadataArrayLength = 100
+	// MaxMetadataDepth is the maximum depth of nested metadata.
+	MaxMetadataDepth = 10
+	// ExcludedMetadataCharactersPattern is the pattern of characters that are not allowed in metadata values.
+	ExcludedMetadataCharactersPattern = "[\\$\\.\\[\\]]"
+	// MaxKeyCount is the maximum number of metadata keys allowed for a run.
+	MaxKeyCount = 1000
+)
+
+// FlattenMetadata flattens a nested map of run metadata into a list of RunMetadataIndex entries.
+func FlattenMetadata(
```

Comment:
@AmanuelAaron just landed a PR that boils down to wanting to index hyperparameters

Is there a way we could only have one way of flattening / indexing JSONB in our code?

Reply:
I would be down to make this more general & push it into a util file that can be shared!